File: src/main/java/org/example/kalkulationsprogramm/domain/MitarbeiterNotiz.java
package org.example.kalkulationsprogramm.domain;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class MitarbeiterNotiz {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(columnDefinition = "TEXT")
    private String inhalt;

    @Column(nullable = false)
    private LocalDateTime erstelltAm;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mitarbeiter_id")
    @JsonBackReference
    private Mitarbeiter mitarbeiter;

    @PrePersist
    public void prePersist() {
        if (erstelltAm == null) {
            erstelltAm = LocalDateTime.now();
        }
    }
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/MitarbeiterNotiz.kt
package org.example.kalkulationsprogramm.domain

import com.fasterxml.jackson.annotation.JsonBackReference
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
open class MitarbeiterNotiz {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(columnDefinition = "TEXT")
    open var inhalt: String? = null

    @Column(nullable = false)
    open var erstelltAm: LocalDateTime? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mitarbeiter_id")
    @JsonBackReference
    open var mitarbeiter: Mitarbeiter? = null

    @PrePersist
    open fun onCreate() {
        if (erstelltAm == null) erstelltAm = LocalDateTime.now()
    }

}